file_paths.md

```diff
@@ -0,0 +1,108 @@
+Puppet Installation Layout
+==========================
+
+This table specifies the file paths in an installation of Puppet and the corresponding Puppet setting, denoted as `:setting`. The 3.x column shows the difference of the specification vs the actual 3.x implementation.
+
+puppet-agent
+------------
+
+    Path                                  Setting                        3.x
+    /etc/puppetlabs                       # :puppetdir                   n/a
```

Comment:
@ahpook @kylog @cprice404 An earlier version of @nwolfe's PR had the `:puppetdir` setting, but I removed that during my refactoring, because it was only mentioned in the confluence doc but not the other tickets. Also I think the name is confusing, a user could easily think `:puppetdir` refers to `/etc/puppetlabs/puppet`, since it is the directory named `puppet`.

I could add the setting, preferably with a different name, and change `:codedir` and `:confdir` to be relative to it?

Or we could just not define a setting for it?